### PR TITLE
frontend: Gracefully handle broken pipe error when highlighting

### PR DESCRIPTION
**Pull request by ryanslade**
We currently handle a few know error types and we've received a support
reqeust from a customer that they're consistently seeing a broken pipe
error for a particular file.

This is most likely caused by a crash on syntect-server so instead of
showing an error to the user we fall back to rendering a non-highlighted
version of the file.

Closes: https://github.com/sourcegraph/customer/issues/594

### CODE REVIEW

File: cmd/frontend/internal/highlight/highlight.go [hunk 1]
```diff
@@ -207,6 +207,9 @@ func Code(ctx context.Context, p Params) (h template.HTML, aborted bool, err err
 		case gosyntect.ErrHSSWorkerTimeout:
 			problem = "hss_worker_timeout"
 		}
+		if problem == "" && strings.Contains(err.Error(), "broken pipe") {
```
Comment (indradhanush) [inline cmd/frontend/internal/highlight/highlight.go:210]:
```suggestion

		if problem == "" && strings.Contains(err.Error(), "broken pipe") {
```

File: cmd/frontend/internal/highlight/highlight.go [hunk 2]
```diff
@@ -207,6 +207,9 @@ func Code(ctx context.Context, p Params) (h template.HTML, aborted bool, err err
...
+			problem = "broken pipe"
+		}
```
Comment (indradhanush) [inline cmd/frontend/internal/highlight/highlight.go:212]:
```suggestion
		}

```

**Review by indradhanush — APPROVED** [review]
Haven't looked at this area of the code before. Changes look good, but it does feel hacky. Not sure how easy / hard it would be to create a custom error type and assert against that type here. 

Suggested changes are nits to add newlines around the code block to separate it from the code blocks before / after it.

**Review by sashaostrikov — APPROVED** [review]
LGTM!

**Comment by ryanslade** [comment]
> Haven't looked at this area of the code before. Changes look good, but it does feel hacky. Not sure how easy / hard it would be to create a custom error type and assert against that type here.

It is a bit hacky, that's true, but we're already doing something similar with the other errors. 

I don't think we can create a custom type here since we're seeing a network failure talking to `syntect-server`, this isn't an error that `syntect-server` can catch and return to us like the others.

**Pull request merged**